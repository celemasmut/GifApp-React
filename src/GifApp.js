import React, {useState} from 'react'
import PropTypes from 'prop-types'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifApp = ( { defaultCategories = [] }) => {
    
    const [categories, setCategories] = useState( defaultCategories )

    // const handleAdd = () => {
    //     setCategories((categ) => [...categ, 'SoyNueva']);
    // }
    
    return (
        <>
            <h2> GifApp </h2>
            <AddCategory setCategories= { setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category  => (
                        <GifGrid
                            key={category} 
                            category = {category}
                        />
                    ))
                }
            </ol>
        </>
    );
}

GifApp.propTypes = {
    inicio: PropTypes.string.isRequired
}

GifApp.defaultProps = {
    inicio : 'Gif-App'
}

export default GifApp

