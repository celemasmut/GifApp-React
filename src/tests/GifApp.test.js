import { shallow } from 'enzyme'
import React from 'react';
import GifApp from '../GifApp';

describe('Prueba de GifApp', () => {
    
    test('debe mostrarse correctamente', () => {

        const wrapper = shallow( <GifApp />);

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe de mostrar una lista de categoria', () => {
        
        const categories = ['Blanck Pink','shakira'];
        const wrapper = shallow( <GifApp defaultCategories={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length);

    })
    
    
})
