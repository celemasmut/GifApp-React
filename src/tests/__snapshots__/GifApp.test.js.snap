// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifApp debe de mostrar una lista de categoria 1`] = `
<Fragment>
  <h2>
     GifApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Blanck Pink"
      key="Blanck Pink"
    />
    <GifGrid
      category="shakira"
      key="shakira"
    />
  </ol>
</Fragment>
`;

exports[`Prueba de GifApp debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
     GifApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
